flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -443,6 +443,35 @@ public ExecutionConfig getExecutionConfig() {
 	}
 
 	/**
+	 * Sets the partitioning of the {@link DataStream} so that the output elements
+	 * are distributed evenly to a subset of instances of the next operation in a round-robin
+	 * fashion.
+	 *
+	 * <p>The subset of downstream operations to which the upstream operation sends
+	 * elements depends on the degree of parallelism of both the upstream and downstream operation.
+	 * For example, if the upstream operation has parallelism 2 and the downstream operation
+	 * has parallelism 4, then one upstream operation would distribute elements to two
+	 * downstream operations while the other upstream operation would distribute to the other
+	 * two downstream operations. If, on the other hand, the downstream operation has parallelism
+	 * 2 while the upstream operation has parallelism 4 then two upstream operations will
+	 * distribute to one downstream operation while the other two upstream operations will
+	 * distribute to the other downstream operations.
+	 *
+	 * <p>In cases where the different parallelisms are not multiples of each other one or several
+	 * downstream operations will have a differing number of inputs from upstream operations.
+	 *
+	 * <p>
```

Comment:
Fine with me :) We would have to make sure that the other docs also mention that it only affects the next operation.